catch-uncommitted.sh

```diff
@@ -33,13 +33,15 @@ done
 
 echo
 
+files="$(git ls-files --exclude-standard --others)"
```

Comment:
There is a test two lines below for the existence of `git` (`echo "git not found, ..."`), so running `git` here before that test seems counterproductive. :-)  Given that some of the `if` blocks end with `exit 0/1`, there's no need for `elif` in all places. Maybe something like:
```bash
if [ $catch_no_git -eq 1 ] && !(hash git 2> /dev/null); then
	echo "git not found, not running catch-uncommitted check"
	exit 0
elif ! git diff HEAD --exit-code ${ignore_eol} -- . "${extra_args[@]}" ; then
	echo '** Uncommitted changes found (^^^ diff above ^^^) - FAIL **'
	exit 1
fi

files="$(git ls-files --exclude-standard --others)"
if test -n "$files" ; then
	echo "$files"
	echo '** Untracked uncommitted files found (^^^ listed above ^^^) - FAIL **'
	exit 2
else
	echo 'No unexpected changes, all good.'
fi


```

Reply:
You're right, I completely forgot about that git existence check! Fixing now...